Minor spill refactors

Ygg01 opened the pull request:
## About the PR 
This is just refactoring some extension methods. No need for changelog.

ElectroJr commented on Content.Server/Fluids/EntitySystems/SpillableSystem.cs:
```diff
@@ -1,33 +1,196 @@
+            : SpillAt(solution, transformComponent.Coordinates, prototype, sound: sound, combine: combine);
+    }
+
+    private void SpillOnLand(EntityUid uid, SpillableComponent component, LandEvent args)
+    {
+        if (args.User != null &&
```
It was like this before, but why is this checking if `user == null`? AFAICT, this means that if an entity is not thrown by a user (e.g., space wind or explosions), then this will not spill. Is that intentional?

ElectroJr replied:
Also, it might be worth adding an admin log here. There are already `LogType.Landed` logs with information about the entity & thrower, so I guess just adding a "[entity] spilled [quantity] reagent on landing" log would probably be enough information. I don't think there is an easy way to summarize the contents of a solution for log?. 

The spill verb will already create a log (although its a generic verb one, with no reagent information).

ElectroJr commented on Content.Server/Fluids/EntitySystems/SpillableSystem.cs:
```diff
@@ -1,33 +1,196 @@
+        if (solution.TotalVolume == 0) return null;
+        
+
+        if (!_mapManager.TryGetGrid(coordinates.GetGridId(EntityManager), out var mapGrid))
+            return null; // Let's not spill to space.
+
+        return SpillAt(mapGrid.GetTileRef(coordinates), solution, prototype, overflow, sound,
+            combine: combine);
```
So turns out that `GetTileRef` can still return space-tiles. I'm not actually sure whether the use of `_mapManager.TryGetGrid(coordinates...` prevents that or not. So maybe still add a `if (TileRef.Tile.IsEmpty)` check just to make sure.

ElectroJr reviewed (approved):
Other than fixing conflicts & maybe adding logs, looks good.

Ygg01 commented:
@ElectroJr Ok, Fixed the conflicts and added some logs, I did ask about what kind of pretty formatting for solution in Discord, but no input and, so I printed everything, as units.

ElectroJr commented:
![logs](https://user-images.githubusercontent.com/60421075/144410465-98b59a0b-4510-404d-830c-318a7b20d6a9.png)

I like the solution `ToString` formatting, and will probably add it to the force-feeding logs. Just two minor niggles: you left a "$" in the log, and the overall log format is worded weirdly. It makes it sound like the thrower is the thing that landed.

I also only realised after testing that if the beaker hits a wall and takes damage, it is the destruction behaviour that causes the spill, not the spill-on-land. Unfortunately the destructible behaviour doesn't have any "who is responsible for this" information so you can't easily add a log there. Though as I think other people suggested some time ago, maybe beakers shouldn't take enough damage from being thrown to shatter, which would also solve this.

Ygg01 commented:
Ok, did the changes, however on my branch the log doesn't appear like in your screenshot
